File: 윤서/그리디/1543.py
# 1543- 문서 검색

docs = input()
word = input()

count = 0
index = 0

while index <= len(docs) - len(word) :      # 검색하려는 길이보다 적게 남으면 멈춤
        if docs[index: index + len(word)] == word :
            count += 1
            index += len(word)              # 검색하고자 하는 워드 수 만큼 index 증가
        else :
            index += 1                      # 찾지 못한 경우 index 1 증가
print(count)
            

File: 찬미/Greedy/1543.swift
var word = readLine()!
var searchWord = readLine()!

var index = 0
var cnt = 0

while word.count - index >= searchWord.count {
    let startIndex = word.index(word.startIndex, offsetBy: index)
    let endIndex = word.index(startIndex, offsetBy: searchWord.count-1)
    if word[startIndex...endIndex] == searchWord {
        cnt += 1
        index += searchWord.count
    } else {
        index += 1
    }
}

print(cnt)
